src/main/java/org/apache/maven/plugins/dependency/tree/VerboseGraphTgfSerializer.java

```diff
@@ -49,7 +49,16 @@ public String serialize( DependencyNode root )
         Artifact rootArtifact = root.getArtifact();
         nodes.append( root.hashCode() ).append( " " ).append( rootArtifact.getGroupId() ).append( ":" ).append(
                 rootArtifact.getArtifactId() ).append( ":" ).append( rootArtifact.getExtension() ).append( ":" ).append(
-                rootArtifact.getVersion() ).append( "\n" );
+                rootArtifact.getVersion() );
+
+        if ( root.getData().containsKey( "ContainsModule" ) )
+        {
+            nodes.append( " WARNING: this tree contains a submodule and once it reaches the submodule will print "
```

Comment:
this is a little hard to follow. a semicolon after the first 'submodule' would help.